src/components/root_layout.tsx

```diff
@@ -16,20 +19,50 @@ interface RootLayoutProps {
  * @remarks
  * Provides a banner that includes a menu
  */
-const RootLayout: FC<RootLayoutProps> = ({children}) => {
+const RootLayout: FC<RootLayoutProps> = ({ children }) => {
+  const location = useLocation();
+  const [templateName, setTemplateName] = useState("");
+  const [jobName, setJobName] = useState("");
+  const [showBackButtonToTemplate, setShowBackButtonToTemplate] = useState(false);
+
+  useEffect(() => {
+    const regexPatternToTemplate = /^.*?\/app\/([^\/]+)\/([^\/]+)$/;
+    const show = regexPatternToTemplate.test(location.pathname);
+    setShowBackButtonToTemplate(show);
+    const matchResult = location.pathname.match(regexPatternToTemplate);
+    if (matchResult) {
+      const [, capturedTemplateName, capturedJobName] = matchResult;
+      setTemplateName(capturedTemplateName);
+      setJobName(capturedJobName);
+    }
+  }, [location.pathname]);
+
+  const showBackButtonToHome = /^.*?\/app\/[^\/]+$/.test(location.pathname);
+
   return (
-    <div style={{marginLeft: "auto", marginRight: "auto", maxWidth: 800, backgroundColor: "rgba(231, 231, 231)"}}>
-      {/* TODO: Add a menu */}
-      <NavBar style={{backgroundColor: "green"}}>
-        <Container>
-          <NavBar.Brand><span style={{color: "gold", fontSize: "2rem"}}>Quality Install Tool</span></NavBar.Brand>
+    <div style={{ marginLeft: "auto", marginRight: "auto", maxWidth: 800, backgroundColor: "rgba(231, 231, 231)" }}>
+      <NavBar style={{ backgroundColor: "green" }}>
+        {showBackButtonToHome && (
```

Comment:
Given the suggested changes above, this can become `showBackButton &&`.

Then I think `<Link to={`app/${templateName}`}` will take the user to correct view. There is no need for separate `showBackButtonToHome` and `showBackButtonToTemplate` sections.